app_tests/integration_tests/llm/conftest.py

```diff
@@ -24,6 +24,20 @@
 logger = logging.getLogger(__name__)
 
 
+def ghstartgroup(msg):
```

Comment:
This and `ghendgroup` may be more shareable with model setup and the actual test if it was in `utils.py`. 

We already import from `utils` in `conftest` and would make it so that we could easily reuse it in the integration test, instead of hardcoding a `::group::/::endgroup::` tag